Content.Server/Decals/DecalSystem.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Content.Shared.Decals;
+using Content.Shared.Maps;
+using Robust.Server.Player;
+using Robust.Shared.Enums;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Player;
+
+namespace Content.Server.Decals
+{
+    public class DecalSystem : SharedDecalSystem
+    {
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+
+        private readonly Dictionary<GridId, HashSet<Vector2i>> _dirtyChunks = new();
+        private readonly Dictionary<IPlayerSession, Dictionary<GridId, HashSet<Vector2i>>> _previousSentChunks = new();
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _playerManager.PlayerStatusChanged += OnPlayerStatusChanged;
+            MapManager.TileChanged += OnTileChanged;
+        }
+
+        public override void Shutdown()
+        {
+            base.Shutdown();
+
+            _playerManager.PlayerStatusChanged -= OnPlayerStatusChanged;
+            MapManager.TileChanged -= OnTileChanged;
+        }
+
+        private void OnTileChanged(object? sender, TileChangedEventArgs e)
+        {
+            if (!e.NewTile.IsSpace())
+                return;
+
+            var chunkCollection = ChunkCollection(e.NewTile.GridIndex);
+            var indices = GetChunkIndices(e.NewTile.GridIndices);
+            var toDelete = new HashSet<uint>();
+            if (chunkCollection.TryGetValue(indices, out var chunk))
+            {
+                foreach (var (uid, decal) in chunk)
+                {
+                    if (new Vector2((int) Math.Floor(decal.Coordinates.X), (int) Math.Floor(decal.Coordinates.Y)) ==
+                        e.NewTile.GridIndices)
+                    {
+                        toDelete.Add(uid);
+                    }
+                }
+            }
+
+            if (toDelete.Count == 0) return;
+
+            foreach (var uid in toDelete)
+            {
+                RemoveDecalInternal(e.NewTile.GridIndex, uid);
+            }
+
+            DirtyChunk(e.NewTile.GridIndex, indices);
+        }
+
+        private void OnPlayerStatusChanged(object? sender, SessionStatusEventArgs e)
+        {
+            switch (e.NewStatus)
+            {
+                case SessionStatus.Connected:
+                    _previousSentChunks[e.Session] = new();
+                    break;
```

Comment:
PVS uses InGame though I'm not sure what the difference is in terms of networking.

I'd probably look at using Connected / InGame as the same one and having default as remove.

Reply:
i should use ingame. its when the player actually exists the lobby and plays the game.